### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by alithethird**
Applicable spec: <link>
Updates docs a lot. Included stories: ISD-2375, ISD-2376, ISD-2377, ISD-2378, ISD-2379, ISD-2385
### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied.
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied.
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`.
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`).
- [x] The changelog is updated with changes that affects the users of the charm.
- [x] The changes do not introduce any regression in code or tests related to LXD runner mode.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Contribute
+
+## Overview
+
+This document explains the processes and practices recommended for contributing enhancements to the GitHub Runner operator.
+
+* Generally, before developing enhancements to this charm, you should consider [opening an issue](https://github.com/canonical/github-runner-operator/issues) explaining your use case.
+* If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).
```
Comment (cbartz) [inline CONTRIBUTING.md:8]:
```suggestion
* If you would like to chat with us about your use-cases or proposed implementation, you can reach us at [Canonical Charm Development Matrix public channel](https://matrix.to/#/#charmhub-charmdev:ubuntu.com) or [Discourse](https://discourse.charmhub.io/).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+This project uses `tox` for managing test environments. There are some pre-configured environments
+that can be used for linting and formatting code when you're preparing contributions to the charm:
+
+* `tox`: Runs all of the basic checks (`lint`, `unit`, `static`, and `coverage-report`).
+* `tox -e fmt`: Runs formatting using `black` and `isort`.
+* `tox -e lint`: Runs a range of static code analysis to check the code.
+* `tox -e static`: Runs other checks such as `bandit` for security issues.
+* `tox -e unit`: Runs the unit tests.
+* `tox -e integration`: Runs the integration tests.
```
Comment (cbartz) [inline CONTRIBUTING.md:38]: It's worth mentioning that this doesn't work locally out of the box, you need an Openstack cloud for some tests and a lot of arguments. So better mention this and remove the line. Even better: You could find out which tests are suitable to run locally and with which arguments (probably most of the LXD suitable tests) and include it here.

File: docs/tutorial/quick-start.md [hunk 3]
```diff
@@ -34,6 +53,11 @@ The registration token can be requested by calling the [GitHub API](https://docs
 
 ### Set up the tutorial model
 
+Switch to the LXD controller:
+```
+juju switch lxd
```
Comment (cbartz) [inline docs/tutorial/quick-start.md:58]: Can you check if this is the default name when bootstrapping lxd? Or maybe mention that it depends on the name you give this controller.

**Review by cbartz — COMMENTED** [review]
Thank you! Looks great. Have some comments.

File: docs/tutorial/quick-start.md (hunk 3 above)
Comment (alithethird) [reply docs/tutorial/quick-start.md:58]: This is the name it gets with the Multipass VM but good point I will add some steps that will read the `juju controller` output and get the lxd name from there

File: CONTRIBUTING.md (hunk 2 above)
Comment (alithethird) [reply CONTRIBUTING.md:38]: Added a note that says: `Integration tests require a lot of arguments, secrets and OpenStack client so they would not work well locally.`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -37,6 +37,7 @@ that can be used for linting and formatting code when you're preparing contribut
 * `tox -e unit`: Runs the unit tests.
 * `tox -e integration`: Runs the integration tests.
 
+> NOTE: Integration tests require a lot of arguments, secrets and OpenStack client so they would not work well locally.
```
Comment (cbartz) [inline CONTRIBUTING.md:40]: Maybe better something like
```suggestion
> NOTE: Integration tests are currently intended to run on the CI and may not work locally without further argument.
```

**Review by cbartz — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Pull request merged**